packages/hardhat-core/src/internal/hardhat-network/provider/utils/random.ts

```diff
@@ -54,4 +54,7 @@ export const randomAddressString = () => {
   return bufferToHex(randomAddressBuffer());
 };
 
-export const randomAddressBuffer = () => randomHashBuffer().slice(0, 20);
+const addressGenerator = RandomBufferGenerator.create("seed");
+export const randomAddressBuffer = (): Buffer => {
+  return addressGenerator.next().slice(0, 20);
+};
```

Comment:
What's the reason for this change?

Reply:
I mimicked the random buffer generator in Rust and let them run side by side for generation of state roots (needed for forked databases, so we don't need to fork all blocks). It turned out that in the tests, the address generator was using the same `RandomBufferGenerator` and thus they would become offset.

I split the addressGenerator so that it has its own seed and won't interfere anymore